UnityProject/Assets/Scripts/Items/Tools/MopTrigger.cs

```diff
@@ -36,13 +39,30 @@ public override bool Interact (GameObject originator, Vector3 position, string h
         return base.Interact (originator, position, hand);
     }
 
-	public void CleanTile (Vector3 spatsPos)
-	{
-		Vector3Int targetWorldIntPos = spatsPos.CutToInt();
-		var bloodSpats = MatrixManager.GetAt<BloodSplat>(targetWorldIntPos);
-		for ( var i = 0; i < bloodSpats.Count; i++ )
-		{
-			bloodSpats[i].DisappearFromWorldServer();
-		}
-	}
+    public void CleanTile (Vector3 worldPos)
+    {
+	    var worldPosInt = worldPos.CutToInt();
+	    var matrix = MatrixManager.AtPoint( worldPosInt );
+	    var localPosInt = MatrixManager.WorldToLocalInt( worldPos, matrix );
```

Comment:
```suggestion
	    var localPosInt = MatrixManager.WorldToLocalInt( worldPosInt, matrix );
```
this fixes it!
for some reason spatsPos has a Z value of -1

Reply:
You're the man.